packages/console/src/pages/Connectors/components/CreateForm/index.tsx

```diff
@@ -21,25 +24,44 @@ type Props = {
 };
 
 const CreateForm = ({ onClose, isOpen: isFormOpen, type }: Props) => {
-  const { data: allGroups, connectors, error } = useConnectorGroups();
-  const isLoading = !allGroups && !connectors && !error;
+  const { data: connectors, error: connectorsError } = useSWR<ConnectorResponse[], RequestError>(
+    '/api/connectors'
+  );
+  const { data: factories, error: factoriesError } = useSWR<
+    ConnectorFactoryResponse[],
+    RequestError
+  >('/api/connector-factories');
+  const isLoading = !factories && !connectors && !connectorsError && !factoriesError;
   const [activeGroupId, setActiveGroupId] = useState<string>();
-  const [activeConnectorId, setActiveConnectorId] = useState<string>();
+  const [activeFactoryId, setActiveFactoryId] = useState<string>();
   const [isGetStartedModalOpen, setIsGetStartedModalOpen] = useState(false);
 
-  const groups = useMemo(
-    () => allGroups?.filter((group) => group.type === type),
-    [allGroups, type]
-  );
+  const groups = useMemo(() => {
+    if (!factories || !connectors) {
+      return [];
+    }
+
+    const allGroups = getConnectorGroups<ConnectorFactoryResponse>(
+      factories.filter(({ type: factoryType }) => factoryType === type)
+    );
+
+    return allGroups.map((group) => ({
+      ...group,
+      connectors: group.connectors.map((connector) => ({
+        ...connector,
+        added: connectors.some(({ connectorId }) => connector.id === connectorId),
```

Comment:
nit: might be a little bit confusing between factory group connectors and DB connectors here.

Reply:
updated with a new name, PTAL